test/unit/org/apache/cassandra/index/sai/cql/types/IndexingTypeSupport.java

```diff
@@ -45,10 +45,31 @@ public abstract class IndexingTypeSupport extends SAITester
 
     public enum Scenario
     {
-        MEMTABLE_QUERY
+        MEMTABLE_QUERY,
+        SSTABLE_QUERY,
+        MIXED_QUERY,
+        COMPACTED_QUERY,
+        POST_BUILD_QUERY
     }
 
-    protected static Collection<Object[]> generateParameters(DataSet<?> dataset)
+    protected static Collection<Object[]> generateLiteralParameters(DataSet<?> dataset)
```

Comment:
Aside: I'm guessing this distinction will later go away when we have on-disk numeric support, etc.?

Reply:
Correct. I wanted, at least, one complete set of tests that would test the on-disk capabilities of the literal index and this seemed the best place to do it. When the numerics are in this will revert to a single method.